src/main/flight/imu.c

```diff
@@ -215,51 +213,79 @@ static float invSqrt(float x)
 STATIC_UNIT_TESTED void imuMahonyAHRSupdate(float dt, float gx, float gy, float gz,
                                 bool useAcc, float ax, float ay, float az,
                                 bool useMag,
-                                float cogYawGain, float courseOverGround, const float dcmKpGain)
+                                float gpsGroundspeedGain, float courseOverGround, const float dcmKpGain)
```

Comment:
?? not clear what you mean ??

The old code always had `courseOverGround` as an input, the sole source of which was GPS.

The old code named the primary (and only) `ez_ef` gain modifier as `cogYawGain`.  In reality, this parameter depended *only* on GPS groundspeed.  All I did here was to re-name that parameter in a way that explain what it does better; this is important if we introduce other `ez_ef` gain modifiers from different sources.

Reply:
This (https://github.com/betaflight/betaflight/blob/master/src/main/flight/imu.c#L545-L557) does not depend 'only on GPS groundspeed' ...

What about `courseOverGroundGain`?

Reply:
However in those lines, it is clear that cogYawGain, other than in a rescue, is only determined by groundspeed:
```
const float imuYawGroundspeed = fminf (gpsSol.groundSpeed / GPS_COG_MAX_GROUNDSPEED, 2.0f);
cogYawGain = (FLIGHT_MODE(GPS_RESCUE_MODE)) ? gpsRescueGetImuYawCogGain() : imuYawGroundspeed;
```
That's why I said that the `cogYawGain` factor hence is actually only a groundspeed factor.
It's not a bad thing to use groundspeed as a factor in the gain factor ez_ef, since GPS heading data is very noisy and inaccurate at low speeds, and since higher speeds are typically in the direction of the nose of the quad.
But we should be clear that this factor is just a number that is zero at zero groundspeed, and rises as groundspeed increases.
Groundspeed is independent of GPS heading; ```
sol->groundSpeed = data->speed;
sol->groundCourse = data->ground_course;
```
where `groundCourse` and `Course Over Ground` refers to angle or heading.

Reply:
Good idea to separate the gain factors clearly.